adapters/beachfront/beachfront.go

```diff
@@ -475,9 +476,11 @@ func getVideoRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraReque
 			}
 		}
 
-		if imp.Video.H == 0 && imp.Video.W == 0 {
-			imp.Video.W = defaultVideoWidth
-			imp.Video.H = defaultVideoHeight
+		if (imp.Video.H == nil || *imp.Video.H == 0) && (imp.Video.W == nil || *imp.Video.W == 0) {
+			videoCopy := *imp.Video
+			videoCopy.W = ptrutil.ToPtr[int64](defaultVideoWidth)
+			videoCopy.H = ptrutil.ToPtr[int64](defaultVideoHeight)
+			imp.Video = &videoCopy
```

Comment:
Beachfront: Fixed unprotected shared memory write. This code path previously had no test coverage.